staker/block_validator.go

```diff
@@ -81,16 +82,18 @@ type BlockValidator struct {
 }
 
 type BlockValidatorConfig struct {
-	Enable                   bool                          `koanf:"enable"`
-	ValidationServer         rpcclient.ClientConfig        `koanf:"validation-server" reload:"hot"`
-	ValidationPoll           time.Duration                 `koanf:"validation-poll" reload:"hot"`
-	PrerecordedBlocks        uint64                        `koanf:"prerecorded-blocks" reload:"hot"`
-	ForwardBlocks            uint64                        `koanf:"forward-blocks" reload:"hot"`
-	CurrentModuleRoot        string                        `koanf:"current-module-root"`         // TODO(magic) requires reinitialization on hot reload
-	PendingUpgradeModuleRoot string                        `koanf:"pending-upgrade-module-root"` // TODO(magic) requires StatelessBlockValidator recreation on hot reload
-	FailureIsFatal           bool                          `koanf:"failure-is-fatal" reload:"hot"`
-	Dangerous                BlockValidatorDangerousConfig `koanf:"dangerous"`
-	MemoryFreeLimit          string                        `koanf:"memory-free-limit" reload:"hot"`
+	Enable                      bool                          `koanf:"enable"`
+	ValidationServer            rpcclient.ClientConfig        `koanf:"validation-server" reload:"hot"`
+	ValidationServerConfigs     []rpcclient.ClientConfig      `koanf:"validation-server-configs" reload:"hot"`
```

Comment:
I don't understand how these configuration options work together. Can you provide some examples in the description? I also don't see where `BlockValidatorConfigAddOptions` adds `ValidationServerConfigs`.

Reply:
`ValidationServerConfigs` as an array of configs cannot be directly handled inside `BlockValidatorConfigAddOptions` like other fields (assigning default value etc..) due to flagSet not supporting such operations hence it is handled inside `Validate` which will be called after all config options are added.